_posts/2023-06-07-airflow-upgrade-to-2-5-3.md

```diff
@@ -0,0 +1,131 @@
+---
+layout: post
+title: "Airflow upgrade from 2.2.0 to 2.5.3 journey"
+author: dimonchik-suvorov
+tags:
+- featured
+- airflow-series
+- aws
+  team: Data Platform
+---
+
+Let me continue QP's `airflow-series` rubric and tell you a story of one Airflow upgrade. In this post I'm going to describe what infrastructure we had before, how we improved it and what difficulties we faced along the way, so let's go!
+
+### At the beginning there were...
+
+Main parts of our Airflow infrastructure:
+- Old but gold Airflow `2.2.0`
+    - Our private fork with custom changes that we didn't upstream yet
+    - Airflow Image based on `python3.7` Docker image stored in AWS ECR (Elastic Container Registry)
+      - We are installing Airflow in `--editable` mode each time to apply our custom changes
+    - Scheduler and Web Server running on AWS ECS Fargate (Elastic Container Service)
+    - Terraform for configuring ECS tasks and Airflow params
+    - Kubernetes Executor on AWS EKS `1.24` (Elastic Kubernetes Service)
+- Airflow backend - MySQL `5.7` on RDS (Relational Database Service)
+![](/post-images/2023-06-07-airflow-upgrade-to-2-5-3/airflow_highlevel_architecture.png)
+<font size="3"><center><i>Databricks cluster configuration parameters</i></center></font>
+
+Also, we have a couple of our own Airflow plugins like custom operators, hooks and _Self-Service Backfill UI_ (you can learn more about this in our [Airflow Summit 2021 slides](https://airflowsummit.org/slides/2021/e5_6-ModernizeAirflow-Scribd.pdf)). 
+And Jenkins pipeline for building Airflow docker image and apply Terraform to restart ECS tasks with new configurations and Airflow image.
+
+### Why?
+
+We decided to upgrade because of released features and fixed bugs obviously, but the most valuable thing we wanted is multiple Schedulers because our Fargate instance came close to it's max available resources, and we suffered from constant high CPU utilization.
+
+This caused MySQL upgrade because multiple Schedulers doesn't work with 5.7 which doesn't have that locking functionality we need...
```

Comment:
Maybe we can add reason explicitly:
```suggestion
This caused MySQL upgrade because multiple Schedulers don't work with MySQL 5.7 which doesn't support SKIP LOCKED or NOWAIT SQL clauses.
```